crypto/context.c

```diff
@@ -158,8 +177,11 @@ int openssl_ctx_init_index(OPENSSL_CTX *ctx, int static_index,
     if (ctx == NULL)
         return 0;
 
-    idx = CRYPTO_get_ex_new_index_ex(ctx, CRYPTO_EX_INDEX_OPENSSL_CTX, 0,
-                                     (void *)meth, openssl_ctx_generic_new,
+    idx = crypto_get_ex_new_index_ex(ctx, CRYPTO_EX_INDEX_OPENSSL_CTX, 0,
+                                     meth == NULL
+                                     ? (void *)methods[static_index]
+                                     : (void *)meth,
```

Comment:
I'd rather that `meth == NULL` be an error.  Considering it's all internal, I think it's justifiable if that leads to a crash, i.e. we should fix the issue before release.

Reply:
The *only* reason "meth == NULL" is here at all is to support the tests. Its the only caller where meth != NULL. As noted above an alternative is pass the method in the openssl_ctx_get_data call.

Reply:
Ah, not originally, but with the latest changes.